doc/en/fixture.rst

```diff
@@ -474,14 +469,17 @@ Here's the ``equipments`` fixture changed to use ``addfinalizer`` for cleanup:
     def equipments(request):
         r = []
         for port in ("C1", "C3", "C28"):
-            equip = connect(port)
-            request.addfinalizer(equip.disconnect)
+            cm = connect(port)
+            equip = cm.__enter__()
+            request.addfinalizer(functools.partial(cm.__exit__, None, None, None))
```

Comment:
:nauseated_face: now this is why ExitStack is so great

Reply:
Oh yeah, for context managers then `ExitStack` is definitely the superior alternative. Do you think it is worth it thought to add this example like this? I'm afraid users might get confused by it.

Reply:
I'm tempted to remove docs for addfinalizer entirely

Reply:
I had a go at removing all pytest's own calls to .addfinalizer and replacing them with plain yield fixtures, and replacing it with either layered resource acquisition or an ExitStack. And it looks much nicer.

It's quite a big change, doesn't really gain anything, and I got bored half way and gave up - but it just demonstrates the superiority of the yield fixture + ExitStack approach.

Reply:
I disagree, it is part of the official API and should be in the docs, even if it is not the best alternative for all cases. People might be maintaining old test suites which use `addfinalizer`, and they might need the docs.

Reply:
FWIW `request.addfinalizer` is also still (AFAIK) the only way to do cleanups from inside a test (rather than a fixture).

Reply:
```python
def test_something():
    with contextlib.ExitStack as stack:
        setup_something()
        stack.callback(cleanup_something)
        assert sut()
```